docs/proposals/VIP_TEMPLATE.md

```diff
@@ -0,0 +1,130 @@
+There is some overlap with the Introduction section but it is here for emphasis since it forms the basis 
+for evaluating any proposed design. There should be no disagreement on the problem statement, 
+otherwise a design or implementation is unlikely to be successful._ 
+
+
+## Scope
+
+_Call out explicitly:_
+
+1. **What is in scope?**
+
+   _For example, this functionality will only apply to batch-only stores._
+
+2. **What is out of scope?**
+
+    _For example, this functionality won’t apply to DaVinci Clients._
+
+## Project Justification
+
+_This section should justify the necessity to address the problem mentioned above, in another way, 
+we could think about what will happen if we don’t solve the problem, and we could consider the following dimensions._
+1. New use cases.
+2. Customer experience
+3. Productivity improvement.
+4. Resilience.
+5. Operability improvement.
+6. Performance improvement.
+7. Maintainability improvement.
+8. Scalability improvement.
+9. Cost to Serve reduction.
+10. Reduce Toil.
+11. Craftsmanship improvement.
+12. Security improvement.
+13. Etc.
+
+## Functional specification
+
+_If this is a development of a functionality/product for users, specify what it should look like from the user's point of view.
+It may include:_
+1. Public API.
+2. Expected public behavior.
+3. Public behavior of error handling, etc.
+
+## Proposed Design
+
+_This section must describe the proposed design and implementation in detail, and how it 
+addresses the problems outlined in the problem statement. It must provide details on alternative solutions 
+that were considered, stating clearly why they lost out to the final design. This latter point is particularly 
+important or else we tend to vacillate on old decisions because we cannot remember the original rationale. 
+Phrase this section in terms of trade offs since it is rare that one approach is better in all ways; there is
+generally a tradeoff and making that explicit makes it much easier to understand._
+
+_The following aspects of the design must be covered when applicable:_
+
+1. _Changes to APIs or protocols or data format highlighting any compatibility issues and how they will be addressed._
+2. _Major components and sequence diagrams between them._
+3. _Multi-region implications including Parent/Child Controller communication._
+4. _Alternative designs considered, and brief explanation of tradeoffs and decision versus the chosen design, 
+and important aspects to consider_:
+   1. Security.
+   2. Performance.
+   3. Maintainability.
+   4. Operability.
+   5. Scalability.
+   6. Compatibility.
+   7. Testability.
+   8. Risks/Mitigations.
+   9. Overall effort.
+   10. Leverageable.
+   11. Align with the long-term vision.
+5. Preferred option by the designer/author.
+   1. _Conclusions/Decisions made in the design review sessions. Explanation why it’s selected_
+
+### Test Plan
```

Comment:
nit: Wondering if this is a big scope design, do we want to add a session or subsession to describe how we can break it down into different phases?

Reply:
Good point. I was thinking the development milestone be covered by the implementation team realm docs but you are right it can be used to sub divide a complex project. Thanks for the suggestion. Added the same.